app/features/notes_generator/core.py

```diff
@@ -0,0 +1,64 @@
+import re
+from typing import Literal, Optional, Dict
+import logging
+from tools import (
+    extract_text_from_file,
+    extract_text_from_url,
+    generate_notes_rag,
+    clean_content
+)
+
+# Dictionary to map input types to corresponding functions
+file_handler_map = {
+    'text': lambda content: clean_content(content),  # Clean content directly
+    'file': extract_text_from_file,
+    'url': extract_text_from_url
+}
+
+# List of supported file extensions
+SUPPORTED_FILE_TYPES = ['.txt', '.docx', '.pdf', '.csv', '.pptx', '.md', '.xml', '.xls', '.xlsx']
+
+def executor(input_type: Literal['text', 'file', 'url'],
+             input_content: Optional[str] = None,
+             file_path: Optional[str] = None,
+             focus_topic: Optional[str] = None,
+             output_structure: Literal['bullet', 'paragraph', 'table'] = 'bullet',
+             export_format: Literal['json'] = 'json') -> Dict[str, str]:
+    try:
+        logging.info("Notes generation started...")
+
+        # Validate that at least one of input_content or file_path is provided
+        if not (input_content or file_path):
+            raise ValueError("Either 'input_content' or 'file_path' must be provided.")
+
+        # Validate the file type if file_path is provided
+        if file_path and not any(file_path.endswith(ext) for ext in SUPPORTED_FILE_TYPES):
+            raise ValueError(f"Unsupported file type for file: {file_path}")
+
+        # Extract content based on input type
+        if input_type in file_handler_map:
+            if input_type == 'text' and input_content:
+                content = file_handler_map[input_type](input_content)
+            elif input_type == 'file' and file_path:
```

Comment:
In the Marvel AI Backend, we are managing files with URLs and specifying the file type. Please, remove this for standardization purposes.